make dependency-check failing

BenediktMKuehne opened the pull request.

m-1-k-3 commented:
As I'm working on another module I have seen that the fail condition is currently working:

![image](https://user-images.githubusercontent.com/497520/147405175-e6b195ac-c038-44a0-aca5-e850e7aeb391.png)

Probably the real issue was line 293: DEP_ERROR=0

BenediktMKuehne commented:
Functionality was/is fine jup.
What this PR changes is the return/exit value from 0 to 1, when using only the dependency check.
Currently you always get 0 on exit even if the check fails.

m-1-k-3 commented:
I am sorry but I don't get it ... probably I misunderstood the issue. could you please explain it in more detail including your testcases.

![image](https://user-images.githubusercontent.com/497520/147953273-39f4cc64-1484-4f56-b265-908862b30b06.png)

BenediktMKuehne commented:
When running `sudo ./emba.sh -d` you get this in the case of any failed check.
![image](https://user-images.githubusercontent.com/62940240/148015394-8573fc1e-0054-41c4-8b06-e2e859ba73cc.png)
Which doesn't let you determine if it failed or succeeded solely by the return value. (since they are the same in both cases)
![image](https://user-images.githubusercontent.com/62940240/148016193-477f491e-178f-4935-81d7-16c53d58036d.png)
But for EMBArk we need this check quite frequently, so I changed the exit conditions.
What I changed is that the dep-check fails on dep-errors and dep-exits and returns `1`.
![image](https://user-images.githubusercontent.com/62940240/148018011-3fc61467-94ce-4cc8-bda7-3b6e9d07a44d.png)

m-1-k-3 commented:
ok, thanks for clarification ... got it now. I will check this PR

m-1-k-3 commented on helpers/helpers_emba_dependency_check.sh:
```diff
@@ -394,7 +394,7 @@ dependency_check()
     print_output "$ORANGE""Learn more about the installation on the emba wiki: ""$NC""https://github.com/e-m-b-a/emba/wiki/installation\\n" "no_log"
 
-    if { [[ $ONLY_DEP -eq 0 ]] && [[ $FORCE -eq 0 ]] ;} || [[ $DEP_EXIT -gt 0 ]] ; then
+    if [[ $ONLY_DEP -eq 1 ]] || [[ $FORCE -eq 0 ]]; then
```
Are you missing the DEP_EXIT now? Shouldn't it be like the following?
if [[ $ONLY_DEP -eq 1 ]] || [[ $FORCE -eq 0 ]] || [[ $DEP_EXIT -gt 0 ]]; then

BenediktMKuehne commented:
Right don't know why I changed that after

m-1-k-3 reviewed (approved):
sorry that it took so long ... thanks for the identification of this bug. Will now merge
